gems/smithy-client/lib/smithy-client/cbor_codec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'base64'
+
+module Smithy
+  module Client
+    # @api private
+    class CBORCodec
```

Comment:
Worth a little discussion - While the aws query compatible *trait* is an AWS specific trait, a `query_compatible` configuration option here is *generic* and as far as codec is concerned, it just means that lists/maps with the [xmlFlattened trait]((https://smithy.io/2.0/spec/protocol-traits.html#xmlflattened-trait) (which is generic, not aws specific) we parse an empty list/map instead of nil.

So - I *think* we could build this behavior in to the codec.  However - I'm also happy to punt that to later - we don't need to implement that now.